### PR TITLE
Add Java callable factory generation

### PR DESCRIPTION
**Pull request by michaelbausor**
Corresponding gax-java PR: https://github.com/googleapis/gax-java/pull/483

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline [hunk 1]
```diff
@@ -6390,6 +6406,100 @@ public class HttpJsonAddressStub extends AddressStub {
...
+
+// AUTO-GENERATED DOCUMENTATION AND CLASS
+/**
+ * HTTP callable factory implementation for simplecompute.
+ *
+ * <p>This class is for advanced usage.
+ */
+@Generated("by GAPIC v0.0.5")
+@BetaApi("The surface for use by generated code is not stable yet and may change in the future.")
+public class AddressCallableFactory implements CallableFactory {
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline]: This naming makes it impossible to add a co-existing CallableFactory that uses Grpc.

Comment (michaelbausor) [reply src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline]: Fixed, PTAL

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java [hunk 2]
```diff
@@ -838,6 +893,35 @@ private void addRpcStubImports(InterfaceContext context) {
...
+    typeTable.saveNicknameFor("com.google.longrunning.stub.OperationsStub");
+
+    switch (((GapicProductConfig) context.getProductConfig()).getTransportProtocol()) {
+      case GRPC:
+        typeTable.saveNicknameFor("com.google.api.gax.grpc.GrpcCallableFactory");
+        typeTable.saveNicknameFor("com.google.api.gax.grpc.CallableFactory");
+        break;
+      case HTTP:
+        typeTable.saveNicknameFor("com.google.api.gax.httpjson.HttpJsonCallableFactory");
+        typeTable.saveNicknameFor("com.google.api.gax.httpjson.CallableFactory");
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java]: We shouldn't use the same class name between multiple packages, given there are cases we might need to reference both in the same file.

Comment (michaelbausor) [reply src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java]: Updated, PTAL

File: src/main/java/com/google/api/codegen/viewmodel/StaticLangCallableFactoryView.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/* Copyright 2017 Google LLC
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/viewmodel/StaticLangCallableFactoryView.java]: 2018 for new files :-P

File: src/main/resources/com/google/api/codegen/java/grpc_callable_factory.snip [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  // AUTO-GENERATED DOCUMENTATION AND CLASS
+  /**
+   * gRPC callable factory implementation for {@doc.serviceTitle}.
+   *
+   * <p>This class is for advanced usage.
+   */
+@end
+
+@private memberMethods(xapiClass)
+  public <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> createUnaryCallable(
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/grpc_callable_factory.snip]: Add `@Override` to overridden methods

**Comment by michaelbausor** [comment]
PTAL

**Pull request merged**